app/src/main/java/fi/bitrite/android/ws/ui/util/NavigationController.java

```diff
@@ -135,38 +142,48 @@ public void navigateToFavoriteUsers() {
         navigateTo(NAVIGATION_TAG_FAVORITE_USERS, FavoriteUsersFragment.create(), true);
     }
 
-    public void navigateToUser(int userId) {
-        navigateTo(NAVIGATION_TAG_MAIN + "/user-" + userId, UserFragment.create(userId), false);
+    public void navigateToContactUser(SimpleUser recipient) {
+        navigateTo(NAVIGATION_TAG_MAIN + "/user-" + recipient.id + "/contact",
+                ContactUserFragment.create(recipient), false);
     }
 
     public void navigateToFeedback(int recipientId) {
         navigateTo(NAVIGATION_TAG_MAIN + "/user-" + recipientId + "/feedback",
                 FeedbackFragment.create(recipientId), false);
     }
 
-    public void navigateToContactUser(SimpleUser recipient) {
-        navigateTo(NAVIGATION_TAG_MAIN + "/user-" + recipient.id + "/contact",
-                ContactUserFragment.create(recipient), false);
+    public void navigateToSearch(String query) {
+        navigateTo(NAVIGATION_TAG_MAIN + "/search-" + query.hashCode(),
+                SearchFragment.create(query), false);
+    }
+
+    public void navigateToUser(int userId) {
+        String tag = NAVIGATION_TAG_MAIN + "/user-" + userId;
+        navigateTo(tag, UserFragment.create(userId), false);
     }
 
     public void navigateToUserList(ArrayList<Integer> userIds) {
         navigateTo(NAVIGATION_TAG_MAIN + "/user_list-" + userIds.hashCode(),
                 UserListFragment.create(userIds), false);
     }
 
-    public void navigateToSearch(String query) {
-        navigateTo(NAVIGATION_TAG_MAIN + "/search-" + query.hashCode(),
-                SearchFragment.create(query), false);
+    interface CreateNewFragmentCallback {
```

Comment:
Can we use `java.util.function.Supplier<Fragment>`?

Reply:
The whole construction is based on the idea of reusingnot only `MapFragment`s, but other fragments as well. Currently, this is not the case, but I'd like to keep it for now.